packages/portal-proto/src/components/Summary/SummaryHeader.tsx

```diff
@@ -19,41 +25,87 @@ export const SummaryHeader = ({
   rightElement,
   isModal = false,
 }: SummaryHeaderProps): JSX.Element => {
+  const [topOffset, setTopOffset] = useState("0px");
+  const isFile = headerTitleLeft === "File";
+  const isProject = headerTitleLeft === "Project";
+
+  useEffect(() => {
+    const globalHeader = document.querySelector("#global-header");
+    const resizeObserver = new ResizeObserver((entries) => {
```

Comment:
Probably can't use mantine's `useResizeObserver` because we need to select an element from outside the component.

Reply:
What is this block of code doing? I wasn't really able to figure it out from looking at the page/code.